### PR TITLE
Feat/comment union find

### PR DESCRIPTION
**Pull request by stevana**

### CODE REVIEW

File: src/Test/StateMachine/Internal/Sequential.hs [hunk 1]
```diff
@@ -255,7 +255,8 @@ checkSequentialInvariant
     monitor $ label s
     pre $ precondition m cmd
     resp <- run $ liftSem (sem m miref) cmd miref
+    let m' = transition m cmd resp
     liftProperty $
-      counterexample ("\nThe post-condition for `" ++ s ++ "' failed!\n\n" ++ show m) $
+      counterexample ("\nThe post-condition for `" ++ s ++ "' failed!\n\n" ++ show m') $
```
Comment (Danten) [inline src/Test/StateMachine/Internal/Sequential.hs]: Hmm? This seems weird to print only the transitioned one.

Comment (stevana) [inline src/Test/StateMachine/Internal/Sequential.hs]: I basically just did it this way to get the desired output for the die hard example.

You reckon we should print both? Something like: "The model was `m`, then `s` happend, post-condition for `s` doesn't hold in the new model `m'`?

Comment (Danten) [inline src/Test/StateMachine/Internal/Sequential.hs]: But the post-condition should hold in the **old** model, not the new.

Comment (stevana) [inline src/Test/StateMachine/Internal/Sequential.hs]: I see what you mean with it being weird to print the transitioned one now, :-).

Hmm...

Comment (Danten) [inline src/Test/StateMachine/Internal/Sequential.hs]: So I still think we can print both, but we need to word it correctly.

**Pull request merged**